novawallet/Common/ViewController/CardLayoutPresentationController/CardLayoutPresentationContoroller.swift

```diff
@@ -0,0 +1,52 @@
+import UIKit
+
+final class CardLayoutPresentationController: UIViewController {
+    // we keep strong reference to delegate pan gesture processing
+    // because translation change is tied with transition.
+    // Delegate doesn't retains controller so we're safe here
+
+    // swiftlint:disable weak_delegate
+    private let transitionDelegate: CardLayoutTransitionDelegateProtocol
```

Comment:
Did you consider to inherit presentation controller from UIPresentationController?